base/system/services/database.c

```diff
@@ -852,6 +852,105 @@ ScmDeleteServiceRecord(PSERVICE lpService)
     DPRINT("Done\n");
 }
 
+DWORD
```

Comment:
I experimented a bit with the (almost last state of this PR):
[Stopping_Services_Works_Pretty_Good_Now_Except_SECLOGON_service.webm](https://github.com/user-attachments/assets/dac03310-e15f-4a2e-9562-8d7988f94a0a)
after SECLOGON failed to be stopped on the first attempt the system gave me no 2nd chance to stop it:
[SECLOGON_gave_me_no_second_chance.webm](https://github.com/user-attachments/assets/14724084-5c0f-4965-8041-732e20f21bbc)
And finally:
[Toggling_DHCP_off_on_off_on_off_on_induced_a_crash.webm](https://github.com/user-attachments/assets/6b4f081c-396e-49a6-98b0-9d3bda36636f)
[Toggling_DHCP_off_on_off_on_off_on_induced_a_crash.log](https://github.com/user-attachments/files/17171061/Toggling_DHCP_off_on_off_on_off_on_induced_a_crash.log)

But even with all those findings I do still think this PR is a clear improvement.